README
weewx communicates with a weather station to produce graphs, reports, and HTML
pages. weewx can publish data to weather services such as WeatherUnderground,
PWSweather.com, or CWOP.

weewx is licensed under the GNU Public License v3.

Documentation: docs/readme.htm or http://weewx.com/docs.html
Wiki: https://github.com/weewx/weewx/wiki
Community support: https://groups.google.com/group/weewx-user


docs/changes.txt
weewx change history
--------------------

4.1.2 MM/DD/YYYY

Fixed problem caused by specifying a [Logging]/[[formatters]] section in
weewx.conf that uses interpolated variables.


4.1.1 06/01/2020

Fixed problem that caused wind speed to be reported to AWEKAS in m/s instead
of km/h.

Fixed problem that caused FTP attempts not to be retried.

Fixed problem that caused monthly and yearly summaries to appear only
sporadically.

Fixed problem when using the ultimeter driver under Python 2.

Fixed problem when using the ws1 driver under Python 2.

Fixed problem that prevented remote directories from being created by FTP.

New strategy for calculating system uptime under Python 3. Revisits
issue #428. Alternative to PR #561.


4.1.0 05/25/2020

Archive records emitted by the Vantage driver now include the number of wind
samples per archive interval in field wind_samples.

wee_import can now import WeatherCat monthly .cat files.

Changed the logging configuration dictionary to match the Python documents.
Thanks to user Graham for figuring this out!

Fixed problem that prevented ws1 driver from working under Python 3. PR #556.

Eliminate use of logging in wee_config, allowing it to be used for installs
without syslog.

Allow expressions to be used as a datatype when plotting.

Added option 'reuse_ssl' to FTP. This activates a workaround for a bug in the
Python ftp library that causes long-lived connections to get closed prematurely.
Works only with Python 3.6 and greater.

The cc3000 driver will automatically reboot the hardware if it stops sending
observations. PR #549.

Install using setup.py forgot to set WEEWX_ROOT when installing in non-standard
places. Fixes issue #546.

Fixed bug in ws28xx driver that prevented it from running under Python 3.
Fixes issue #543.

Changed query strategy for calculating min and max wind vectors, which
should result in much faster plot generation.

Fixed bug in wmr9x8 driver that prevented it from running under Python 3.

Fixed several bugs in the te923 driver that prevented it from running under
Python 3.

Added a logging handler for rotating files. See https://bit.ly/2StYSHb for how
to use it. It is the default for MacOS.

More information if an exception is raised while querying for vantage hardware
type.

wunderfixer: fixed problem under Python 3 where response was not converted to
str before attempting to parse the JSON. Option --simulate now requires api_key
and password, so it can hit the WU.

Fixed problem in te923 driver under Python 3 that caused it to crash.


4.0.0 04/30/2020

Ported to Python 3. WeeWX should now run under Python 3.5 and greater, as well
as Python 2.7. Support for Python 2.5 and 2.6 has been dropped.

New facility for creating new user-defined derived types. See the Wiki article
https://github.com/weewx/weewx/wiki/WeeWX-V4-user-defined-types

WeeWX now uses the Python 'logging' facility. This means log, formats, and other
things can now be customized. Fixes issue #353.

Strings appearing in the data stream no longer cause a TypeError if they can be
converted to a number.

Strings can now be accumulated and extracted in the accumulators, making it
possible to include them in the database schemas.

The utility wee_reports now loads services, allowing it to use user-supplied
extensions. Fixes issue #95.

New default schema ("wview_extended") that offers many new types. The old schema
is still supported. Fixes issue #115.

Optional, more flexible, way of specifying schemas for the daily summaries. The
old way is still supported.

The install process now offers to register a user's station with weewx.com.

The package MySQL-python, which we used previously, is not always available on
Python 3. Ported the MySQL code to use the package mysqlclient as an
alternative.

The default for WOW no longer throttles posting frequency (the default used to
be no more than once every 15 minutes).

Added new aggregate types minsum, minsumtime, sum_le. PR #382.

Unit group group_distance is now a first-class group.

Added new tag $python_version.

Ported to Python2-PyMySQL package on OpenSUSE.

Added new aggregation types 'first' (similar to 'last'), 'diff' (the difference
between last and first value in the aggregation interval), and 'tderiv' (the
difference divided by the time difference).

Created new unit group 'group_energy2', defined as watt-seconds. Useful for high
resolution energy monitors.

An observation type known to the system, but not in a record, will now return a
proper ValueTuple, rather than UnknownType.

Type "stormStart" was added to the unit system. Fixes issue #380.

Added new aggregation type 'growdeg'. Similar to 'heatdeg', or 'cooldeg', it
measures growing degree-days. Basically, cooldeg, but with a different base.
Fixes issue #367. Thanks to user Clay Jackson for guidance!

Ported OS uptime to OpenBSD. Fixes issue #428. Thanks to user Jeff Ross!

Catch SSL certificate errors in uploaders. Retry after an hour. Fixes issue
#413.

Wunderfixer has been ported to the new WU API. This API requires an API key,
which you can get from the WU. Put it in weewx.conf. Added option --upload-only.
Thanks to user Leon Shaner! Fixes issues #414 and #445.

Wee_import can now import Weather Display monthly log files.

Fixed problem where sub-sections DegreeDays and Trend were located under the
wrong weewx.conf section. Fixes issue #432. Thanks to user mph for spotting
this!

Added new parameters to the Weather Underground uploader. Fixes issue #435.

Added new air quality types pm1_0, pm2_5, and pm10_0 to the unit system. Added
new unit microgram_per_meter_cubed. Added new unit group, group_concentration.

Plist for the Mac launcher now includes a log file for stderr.

Night-day transition in plots now uses shortest travel distance around color

pkg/changelog.el
* Sat May 30 2020 Matthew Wall (weewx) <mwall@users.sourceforge.net> - 4.1.1-1
- new upstream release
- remove the implicitly-applied dependencies in the redhat rpms
* Mon May 25 2020 Matthew Wall (weewx) <mwall@users.sourceforge.net> - 4.1.0-1
- new upstream release
* Thu Apr 30 2020 Matthew Wall (weewx) <mwall@users.sourceforge.net> - 4.0.0-1
- new upstream release
* Thu Apr 09 2020 Matthew Wall (weewx) <mwall@users.sourceforge.net> - 4.0.0b18-2
- fix python/python3 invocations
* Thu Apr 09 2020 Matthew Wall (weewx) <mwall@users.sourceforge.net> - 4.0.0b18-1
- new upstream release
* Tue Mar 31 2020 Matthew Wall <mwall@users.sourceforge.net> - 4.0.0b17-1
- new upstream release
* Wed Feb 26 2020 Matthew Wall <mwall@users.sourceforge.net> - 4.0.0b13-1
- new upstream release
* Sun Feb 02 2020 Matthew Wall <mwall@users.sourceforge.net> - 4.0.0b11-1
- new upstream release
* Sat Jan 04 2020 Matthew Wall <mwall@users.sourceforge.net> - 4.0.0b6-1
- new upstream release
* Sun Jul 14 2019 [ultimate] Matthew Wall (weewx) <mwall@users.sourceforge.net> - 3.9.2-1
- new upstream release
* Wed Feb 06 2019 Matthew Wall (weewx) <mwall@users.sourceforge.net> - 3.9.1-2
- fix html_root location for suse
* Wed Feb 06 2019 Matthew Wall (weewx) <mwall@users.sourceforge.net> - 3.9.1-1
- new upstream release
* Tue Feb 05 2019 Matthew Wall (weewx) <mwall@users.sourceforge.net> - 3.9.0-1
- new upstream release
* Mon Jan 28 2019 Matthew Wall (weewx) <mwall@users.sourceforge.net> - 3.9.0b3-1
- new upstream release
* Sat Jan 26 2019 Matthew Wall (weewx) <mwall@users.sourceforge.net> - 3.9.0b2-1
- new upstream release
* Tue Jan 22 2019 Matthew Wall (weewx) <mwall@users.sourceforge.net> - 3.9.0b1-1
- new upstream release
* Thu Aug 16 2018 Matthew Wall (weewx) <mwall@users.sourceforge.net> - 3.8.2-1
- new upstream release
* Fri Jun 22 2018 Matthew Wall (weewx) <mwall@users.sourceforge.net> - 3.8.1-1
- new upstream release
* Tue Nov 21 2017 mwall <mwall@users.sourceforge.net> - 3.8.0-1
- new upstream release
* Tue Nov 21 2017 mwall <mwall@users.sourceforge.net> - 3.8.0a2-1
- new upstream release
* Wed Mar 22 2017 Matthew Wall (weewx) <mwall@users.sourceforge.net> - 3.7.1-1
- new upstream release
* Fri Mar 10 2017 Matthew Wall (weewx) <mwall@users.sourceforge.net> - 3.7.0-1
- new upstream release
* Sat Mar 04 2017 Matthew Wall (weewx) <mwall@users.sourceforge.net> - 3.7.0b3-1
- new upstream release
* Sat Feb 18 2017 Matthew Wall (weewx) <mwall@users.sourceforge.net> - 3.7.0b2-1
- new upstream release
* Thu Feb 09 2017 Matthew Wall (weewx) <mwall@users.sourceforge.net> - 3.7.0a3-1
- new upstream release
* Thu Oct 13 2016 Matthew Wall (weewx) <mwall@users.sourceforge.net> - 3.6.1-1
- new upstream release
* Fri Oct 07 2016 Matthew Wall (weewx) <mwall@users.sourceforge.net> - 3.6.0-1
- new upstream release
* Tue Oct 04 2016 Thomas Keffer (Author of weewx) <tkeffer@gmail.com> - 3.6.0b3-1
- new upstream release
* Mon Sep 26 2016 Matthew Wall (weewx) <mwall@users.sourceforge.net> - 3.6.0b2-1
- new upstream release
* Sun Sep 25 2016 Matthew Wall (weewx) <mwall@users.sourceforge.net> - 3.6.0b1-1
- new upstream release
* Thu Sep 22 2016 Matthew Wall (weewx) <mwall@users.sourceforge.net> - 3.6.0a1-1
- new upstream release
* Sun Mar 13 2016 Matthew Wall (weewx) <mwall@users.sourceforge.net> - 3.5.0-1
- new upstream release
* Sat Jan 16 2016 Matthew Wall (weewx) <mwall@users.sourceforge.net> - 3.4.0-1
- new upstream release
* Sun Dec 06 2015 Matthew Wall (weewx) <mwall@users.sourceforge.net> - 3.3.1-1
- new upstream release
* Sat Dec 05 2015 Matthew Wall (weewx) <mwall@users.sourceforge.net> - 3.3.0-1
- new upstream release
* Sat Oct 31 2015 mwall <mwall@picodeb8> - 3.3.0b1-1
- new upstream release
* Sat Jul 18 2015 Matthew Wall (weewx) <mwall@users.sourceforge.net> - 3.2.1-1
- new upstream release
* Wed Jul 15 2015 Matthew Wall (weewx) <mwall@users.sourceforge.net> - 3.2.0-1
- new upstream release
* Wed Jul 15 2015 Thomas Keffer (Author of weewx) <tkeffer@gmail.com> - 3.2.0-1
- new upstream release
* Wed Jul 15 2015 Matthew Wall (weewx) <mwall@users.sourceforge.net> - 3.2.0b3-1
- new upstream release
* Tue Jul 07 2015 Matthew Wall (weewx) <mwall@users.sourceforge.net> - 3.2.0b1-1
- new upstream release
- fixes to rpm pre/post scripts in weewx.spec
* Sun Jul 05 2015 Matthew Wall (weewx) <mwall@users.sourceforge.net> - 3.2.0a4-1
- new upstream release
* Sat Apr 25 2015 Matthew Wall (weewx) <mwall@users.sourceforge.net> - 3.2.0a1-1
- new upstream release
- use unified wee_X utilities
* Thu Feb 05 2015 Matthew Wall (weewx) <mwall@users.sourceforge.net> - 3.1.0-1
- new upstream release
* Sat Dec 06 2014 Matthew Wall (weewx) <mwall@users.sourceforge.net> - 3.0.1-1
- new upstream release
* Fri Dec 05 2014 Thomas Keffer (Author of weewx) <tkeffer@gmail.com> - 3.0.0-1
- new upstream release
* Mon Dec 01 2014 Matthew Wall (weewx) <mwall@users.sourceforge.net> - 3.0.0b2-1
- new upstream release
* Sat Nov 29 2014 Matthew Wall (weewx) <mwall@users.sourceforge.net> - 3.0.0a5-1
- new upstream release
* Sat Nov 29 2014 Matthew Wall (weewx) <mwall@users.sourceforge.net> - 3.0.0a4-1
- new upstream release
* Fri Nov 28 2014 Matthew Wall (weewx) <mwall@users.sourceforge.net> - 3.0.0a3-1
- new upstream release
* Thu Nov 27 2014 Matthew Wall (weewx) <mwall@users.sourceforge.net> - 3.0.0a2-1
- new upstream release
* Sat Oct 11 2014 Matthew Wall (weewx) <mwall@users.sourceforge.net> - 2.7.0-1
- new upstream release
* Mon Jun 16 2014 Matthew Wall (weewx) <mwall@users.sourceforge.net> - 2.6.4-1
- new upstream release
- added cc3000, ultimeter, ws1 drivers
* Thu Apr 10 2014 Matthew Wall (weewx) <mwall@users.sourceforge.net> - 2.6.3-1
- new upstream release
* Sun Feb 16 2014 Matthew Wall (weewx) <mwall@users.sourceforge.net> - 2.6.2-1
- new upstream release
* Sat Feb 08 2014 Matthew Wall (weewx) <mwall@users.sourceforge.net> - 2.6.1-1
- new upstream release
* Fri Feb 07 2014 Matthew Wall (weewx) <mwall@users.sourceforge.net> - 2.6.0-1
- new upstream release
* Wed Feb 05 2014 Matthew Wall (weewx) <mwall@users.sourceforge.net> - 2.6.0b2-1
- new upstream release
* Tue Feb 04 2014 Matthew Wall (weewx) <mwall@users.sourceforge.net> - 2.6.0b1-1
- new upstream release
* Tue Jan 28 2014 Matthew Wall (weewx) <mwall@users.sourceforge.net> - 2.6.0a6-1
- new upstream release
* Mon Dec 30 2013 Matthew Wall <mwall@users.sourceforge.net> - 2.5.1-1
- added ws23xx and te923 drivers
* Tue Oct 29 2013 Matthew Wall <mwall@users.sourceforge.net> - 2.5.0-1
- new upstream release
* Sat Oct 19 2013 Matthew Wall <mwall@users.sourceforge.net> - 2.5.0b3-1
- new upstream release
* Fri Oct 18 2013 Matthew Wall <mwall@users.sourceforge.net> - 2.5.0b2-1
- new upstream release
* Sat Oct 12 2013 Matthew Wall <mwall@users.sourceforge.net> - 2.5.0b1-1
- new upstream release
* Sun Aug 04 2013 Matthew Wall <mwall@users.sourceforge.net> - 2.4.0-1
- new upstream release
* Sat Jun 22 2013 Matthew Wall <mwall@users.sourceforge.net> - 2.3.3-1
- new upstream release
* Sun Jun 16 2013 Matthew Wall <mwall@users.sourceforge.net> - 2.3.2-1
- new upstream release
* Mon Apr 15 2013 Thomas Keffer <tkeffer@gmail.com> - 2.3.1-1
- new upstream release
* Tue Apr 09 2013 Matthew Wall <mwall@users.sourceforge.net> - 2.3.0-1
- new upstream release
* Fri Feb 15 2013 Matthew Wall <mwall@users.sourceforge.net> - 2.2.1-1
- fixed ordinals
* Thu Feb 14 2013 Matthew Wall <mwall@users.sourceforge.net> - 2.2.0-1
- no packaging changes for the 2.2.0 release
* Wed Feb 13 2013 Matthew Wall <mwall@users.sourceforge.net> - 2.2.0b2-1
- second beta for 2.2.0
* Sun Feb 10 2013 Matthew Wall <mwall@users.sourceforge.net> - 2.2.0b1-1
- first beta for 2.2.0
* Sat Feb 09 2013 Matthew Wall <mwall@users.sourceforge.net> - 2.2.0a5-1
- fixed postrm to work with ubuntu systems
* Fri Feb 08 2013 Matthew Wall <mwall@users.sourceforge.net> - 2.2.0a4-1
- include logrotate and syslog snippets
- use wee_config_* naming
* Sun Feb 03 2013 Matthew Wall <mwall@users.sourceforge.net> - 2.2.0a3-1
- removed apache dependencies
- put generated html in /var/www/html/weewx
* Mon Jan 28 2013 Matthew Wall <mwall at users.sourceforge.net> - 2.2.0a2-1
- merged packaging branch to trunk
- put chkconfig in preun rather than postun as per rpmlint suggestion
* Sat Jan 26 2013 Matthew Wall <mwall at users.sourceforge.net> - 2.1.1-1
- initial redhat package
